src/core/Akka.Persistence/Persistence.cs

```diff
@@ -99,19 +95,17 @@ public IActorRef SnapshotStoreFor(string snapshotPluginId)
         public IActorRef JournalFor(string journalPluginId)
         {
             var configPath = string.IsNullOrEmpty(journalPluginId) ? _defaultJournalPluginId.Value : journalPluginId;
-            Lazy<PluginHolder> pluginContainer;
-            var extensionIdMap = _journalPluginExtensionIds.Value;
-            if (!extensionIdMap.TryGet(configPath, out pluginContainer))
-            {
-                pluginContainer = new Lazy<PluginHolder>(() => CreatePlugin(configPath, type =>
-                    typeof (AsyncWriteJournal).IsAssignableFrom(type)
```

Comment:
It should be thread safe. CompareAndSet Sets only the new value if extensionIdMap is still the same. JournalFo rshould return allways the valid value